source/connection_manager.c

```diff
@@ -586,7 +642,7 @@ static void prvMQTTAgentTask( void * pvParameters )
 {
     BaseType_t xNetworkResult = pdFAIL;
     MQTTStatus_t xMQTTStatus = MQTTSuccess;
-    MQTTContext_t * pMqttContext = NULL;
+    MQTTContext_t * pMqttContext = &( xGlobalMqttAgentContext.mqttContext );
```

Comment:
Why is xGlobalMqttAgentContext used instead of pMqttAgentContext?

Reply:
This is a demo file, so the agent context is declared here